app/services/permissions/order.rb

```diff
@@ -7,17 +7,25 @@ def initialize(user)
 
     # Find orders that the user can see
     def visible_orders
-      Spree::Order.where(id:
-        managed_orders.select(:id) |
-        coordinated_orders.select(:id) |
-        produced_orders.select("spree_orders.id"))
+      Spree::Order.
+        with_line_items_variants_and_products_outer.
+        where(
+          Spree::Order.arel_table.
+            # Grouping keeps the 2 where clauses from produced_orders_where_values inside parentheses
+            #   This way it makes the OR work between the 3 types of orders:
+            #     produced, managed and coordinated
+            grouping(produced_orders_where_values).
+            or(managed_orders_where_values).
+            or(coordinated_orders_where_values)
+        )
```

Comment:
IMO it'd be worth splitting this class into smaller purpose-specific classes like `visible_orders`. Even if only used within this `*_orders` methods, it'd give us room for comments on extracted private methods like the one above. Up to you. Just suggesting.

Reply:
I am always up for some extraction :-)
in this case though I prefer to keep it together because, we are at around 100lines for the class, still ok, and the methods are very closely related: in #4551 you see the line_items query uses the orders query where clauses. Maybe on the next iteration, ok?

Reply:
I think we will naturally end up with multiple classes as we evolve this crazy queries